spec/jsonls_spec.lua

```diff
@@ -0,0 +1,222 @@
+        schema = {
+          type = 'object',
+          properties = {
+            customField = { type = 'string' },
+          },
+        },
+      }
+
+      vim.lsp.config('jsonls', {
+        settings = {
+          json = {
+            schemas = { existing_schema },
+          },
+        },
+      })
+
+      -- Act: Get schemas (should include existing ones)
+      local schemas = Jsonls.get_json_schemas()
+
+      -- Assert: The existing schema should be first (preserved)
+      assert.is_not_nil(schemas)
+      assert.is_true(#schemas > 0)
+      assert.same(existing_schema, schemas[1])
+    end)
+
+    it('should not mutate the original existing_schemas table', function()
+      -- Setup: Configure jsonls with existing schemas
+      local original_schema = {
+        fileMatch = { 'original.json' },
+        schema = { type = 'object' },
+      }
+      local existing_schemas = { original_schema }
+
+      vim.lsp.config('jsonls', {
+        settings = {
+          json = {
+            schemas = existing_schemas,
+          },
+        },
+      })
+
+      local original_length = #existing_schemas
+
+      -- Act: Get schemas
+      local schemas = Jsonls.get_json_schemas()
+
+      -- Assert: Original table should not be modified
+      assert.equal(original_length, #existing_schemas)
+      assert.same({ original_schema }, existing_schemas)
+      -- The returned schemas should have more items (original + plugin schemas)
+      assert.is_true(#schemas >= original_length)
+    end)
+
+    it('should cache the result', function()
+      -- First call
+      local schemas1 = Jsonls.get_json_schemas()
+
+      -- Second call should return the same cached instance
+      local schemas2 = Jsonls.get_json_schemas()
+
+      assert.equal(schemas1, schemas2)
+    end)
+
+    it('should reload schemas after cache is cleared', function()
+      -- First call
+      local schemas1 = Jsonls.get_json_schemas()
+
+      -- Clear cache
+      Jsonls.clear_cache()
+
+      -- Second call should return a new instance
+      local schemas2 = Jsonls.get_json_schemas()
+
+      assert.is_not.equal(schemas1, schemas2)
+      -- But content should be the same
+      assert.same(schemas1, schemas2)
+    end)
+
+    it('should handle empty existing schemas', function()
+      -- Setup: No existing schemas
+      vim.lsp.config('jsonls', {
+        settings = {
+          json = {
+            schemas = {},
+          },
+        },
+      })
+
+      -- Act
+      local schemas = Jsonls.get_json_schemas()
+
+      -- Assert: Should still return plugin schemas
+      assert.is_not_nil(schemas)
+      assert.is_table(schemas)
+    end)
+
+    it('should handle missing jsonls config', function()
+      vim.lsp.config.jsonls = {}
+      local schemas = Jsonls.get_json_schemas()
+
+      -- Assert: Should return plugin schemas
+      assert.is_not_nil(schemas)
+      assert.is_table(schemas)
+    end)
+  end)
+
+  describe('setup', function()
```

Comment:
I think we can drop this whole block, the interesting logic all happens in `get_schemas`, and there is no way to clear LSP configs in Neovim in between tests so the purity of the tests is questionable.